src/ecs/rendering/resources.rs

```diff
@@ -135,3 +137,33 @@ impl AsRef<Rgba> for AmbientColor {
         &self.0
     }
 }
+
+/// This specs resource with id 0 permits sending commands to the
+/// renderer internal window.
+pub struct WindowCommander {
+    // It's unlikely we'll get more than two commands per frame
+    // 2 Boxes also make this the same size as a Vec, so this costs
+    // no more space in the structure than a Vec would.
+    //
+    // NOTE TO FUTURE AUTHORS: This could be an FnOnce but that's not possible
```

Comment:
I solved this in https://github.com/slide-rs/specs/pull/214/files , maybe you can use that.

Reply:
Ha!  Not bad!  I like it.  I'll give it a shot.